internal/platform/open_file_windows.go

```diff
@@ -106,9 +106,6 @@ func open(path string, mode int, perm uint32) (fd syscall.Handle, err error) {
 	case syscall.O_RDWR:
 		access = syscall.GENERIC_READ | syscall.GENERIC_WRITE
 	}
-	if mode&syscall.O_CREAT != 0 {
-		access |= syscall.GENERIC_WRITE
-	}
```

Comment:
@codefromthecrypt I had to make this change to have the test pass on windows. I'm not sure I understand why this was needed, since you introduced this code initially you may have more context to share?

Reply:
so the godoc on this code says it was originally from go, but actually it has drifted since, which I guess is a good thing. We should probably make comments under the link about exceptions.

```
// # Differences from syscall.Open
//
// This code is based on syscall.Open from the below link with some differences
// https://github.com/golang/go/blame/go1.20/src/syscall/syscall_windows.go#L308-L379
//
//   - syscall.O_CREAT doesn't imply syscall.GENERIC_WRITE as that breaks flag expectations
//      in wasi.
```

This was the change in go over a decade ago which is now gone (fine with me) https://github.com/golang/go/commit/035265975fbf93fa67aa6a6e7611ba640691d693